homeworks/veizerattila/2_vars_datatypes/user_info.py

```diff
@@ -0,0 +1,119 @@
+﻿# Feladat 2: List és dictionary műveletek használata
+####################################################
+import pprint as pp
+
+# Dictionary létrehozása:
+user_info = {
+    "name": "Mike",
+    "age": 25,
+    "favourite_meals": [
+        "pizza",
+        "carbonara",
+        "sushi"
+    ],
+    "phone_contacts": {
+        "Mary": "+36701234567",
+        "Tim": "+36207654321",
+        "Tim2": "+36304567321",
+        "Jim": "+364005000"
+    }
+}
+
+# 1. Kérj be a felhasználótól 4 programozási nyelvet vesszővel elválasztva, szóközök nélkül. Konvertáld a kapott stringet egy listává, és add hozzá a fenti dictionary-hez “skills” néven.
+print("1. Kérj be a felhasználótól 4 programozási nyelvet vesszővel elválasztva, szóközök nélkül.")
+print("Konvertáld a kapott stringet egy listává, és add hozzá a fenti dictionary-hez “skills” néven!")
+print("==========================================================================================================")
+prog_skills = input("Give me 4 programming languages you're familiar with, separated by a comma! ")
+
+if prog_skills.count(",")  == 3:
+# Az if igaz ága csak akkor enged át, ha valaki 4 inputot adott meg, 3 vesszővel elválasztva
+
+    skills = {"skills": prog_skills.split(",")}
+    user_info.update(skills)
+
+    pp.pprint(user_info)
+    print("==========================================================================================================")
+
+
+    # 2. Rendezd a favourite_meals lista elemeit abc szerinti növekvő sorrendbe.
+    keys = list(user_info.keys()) # kulcs változó definiálása, hogy így a dict. elemeire a key alapján k;s[bbi lehessen hivatkozni (pl.: "favourite_meals" helyett "2")
+    user_info[keys[2]] = sorted(user_info[keys[2]])
+    print("2. Rendezd a favourite_meals lista elemeit abc szerinti növekvő sorrendbe!")
+    pp.pprint(user_info)
+    print("==========================================================================================================")
+
+
+    # 3. Printeld ki a favourite_meals lista utolsó előtti elemét
+    print("3. Printeld ki a favourite_meals lista utolsó előtti elemét!")
+    pp.pprint(f"A 'favourite_meals' lista utolsó előtti eleme: {user_info[keys[2]][-2]}")
+    print("==========================================================================================================")
+
+
+    # 4. Adj hozzá egy “spaghetti” string-et ugyanehhez a listához.
+    user_info[keys[2]].append("spaghetti")
+    print("4. Adj hozzá egy “spaghetti” string-et ugyanehhez a listához!")
+    pp.pprint(user_info)
+    print("==========================================================================================================")
+
+
+    # 5. Add hozzá a favourite_meals-hez az aktuális favourit_meals lista harmadik és negyedik elemét (nem az index-ét) újra.
+    user_info[keys[2]].extend(user_info[keys[2]][2:3])
```

Comment:
Szia, ez így már majdnem jó, csak ugye a range 2. tagja exclusive, tehát nem fog beletartozni a kiválasztásba. Így a [2:3] csak a 3. elemet adja vissza, a 4.-et nem. [2:4] lesz a megoldás :)

Reply:
Szia István, javítottam a fenti, további két jelzés, teljesen jogosak és tanulok én is belőle :) Elnézést a figyelmetlenségért és köszönöm a visszajelzést!